tests/check_check_tags.c

```diff
@@ -0,0 +1,315 @@
+/*
+ * Check: a unit test framework for C
+ * Copyright (C) 2001, 2002 Arien Malec
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+#include "../lib/libcompat.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <check.h>
+#include <assert.h>
+#include "check_check.h"
+
+/* from check_check_master.c */
```

Comment:
It may be better to add these tests to the same places where the other tests are located. Try in check_check_selective.c. Check allows a unit test to create suites and run unit tests. This is how Check tests itself. You may need to do this in check_check_selective.c.

What may get you is that all the suites which are added in check_check_main.c have an entry in check_check_master.c in the following table:
```
static master_test_t master_tests[]
```
where the entry specifies the suite name, the expected result, and the expected output. For example:
```
{ "Simple Tests", CK_FAILURE, "This test should fail" },
```
Each suite is listed in the order they are added, then for each suite the tests are in the order they were added.

Hopefully by adding the tests to the master list some of the duplicated code from check_check_master.c can be avoided.